libs/cardano-ledger-core/src/Cardano/Ledger/CompactAddress.hs

```diff
@@ -229,12 +230,10 @@ getPayCred header = case testBit header payCredIsScript of
   True -> getScriptHash
   False -> getKeyHash
 
--- | WARNING: This optimized version of isBootstrapRedeemer does not agree
--- with the one in Cardano.Ledger.Address
-isBootstrapRedeemer :: CompactAddr crypto -> Bool
-isBootstrapRedeemer (UnsafeCompactAddr bytes) =
-  testBit header byron -- AddrBootstrap
-    && addrType == 2 -- ATRedeem
+-- | Optimized version of `Address.isBootstrapRedeemer`, where we avoid
+-- decompacting of post Byron addresses.
+isBootstrapRedeemer :: forall crypto. CC.Crypto crypto => CompactAddr crypto -> Bool
+isBootstrapRedeemer cAddr@(UnsafeCompactAddr bytes) =
+  testBit header byron && Address.isBootstrapRedeemer (decompactAddr cAddr :: Addr crypto)
```

Comment:
i guess this still avoids decompacting the addr more often then not

Reply:
Yep, that's the point. There aren't too many Byron addresses, when compared to the whole UTxO

Reply:
this function is actually only ever called on the epoch boundary when Shelley transitions to Allegra. At that time, there actually were/are probably still a lot of Byron addresses.